ui/next/app/containers/banner/helpusBanner.tsx

```diff
@@ -0,0 +1,86 @@
+/// <reference path="../../../typings/index.d.ts" />
+import * as React from "react";
+import { connect } from "react-redux";
+import { Link, RouterOnContext } from "react-router";
+import _ = require("lodash");
+
+import Banner from "./banner";
+import { KEY_HELPUS, OptInAttributes, KeyValue, loadUIData, saveUIData } from "../../redux/uiData";
+
+class HelpusBannerProps {
+  attributes: OptInAttributes;
+  attributesLoaded: boolean;
+  loadUIData: (...keys: string[]) => void;
+  saveUIData: (...values: KeyValue[]) => void;
+}
+
+class HelpusBannerState {
+  dismissed: boolean = false;
+}
+
+interface HelpusBannerContext {
+  router: RouterOnContext;
+}
+
+class HelpusBanner extends React.Component<HelpusBannerProps, HelpusBannerState> {
+  static contextTypes: React.ValidationMap<any> = {
+    router: React.PropTypes.object,
+  };
+
+  state = new HelpusBannerState();
+  context: HelpusBannerContext;
+
+  componentWillMount() {
+    if (this.context && this.context.router && this.context.router.isActive("/help-us/reporting")) {
```

Comment:
I guess I still don't fully understand the rule of thumb here. I understood why we stored the table sort order in the local ui data reducer object, because as the table is destroyed/recreated we want to preserve the sort.

But this feels so much more self-contained to me. There's only one helpus banner and it's active the entire time. Why should the routing system know about the helpus banner and dismissing it?